src/object/get.js

```diff
@@ -41,12 +41,17 @@ module.exports = (send) => {
 
     send({
       path: 'object/get',
-      args: cidB58Str
+      args: cidB58Str,
+      qs: {
+        'data-encoding': 'base64'
```

Comment:
@alanshaw  quick question: what values are allowed here? 
Ones from https://github.com/multiformats/multibase/blob/master/multibase.csv ?

Reply:
At the moment `go-ipfs` [supports `text` and `base64`](https://github.com/ipfs/go-ipfs/blob/master/core/commands/object/object.go#L568-L579).

Go's `text` is the default (e.g. raw bytes), though `go-ipfs` doesn't escape the characters so it's http api can produce strings that are invalid in JSON and hard to replicate in JavaScript so `base64` is safer.

Once ipfs/js-ipfs#1420 is in, `js-ipfs` will support whatever [Buffer.toString([encoding[, start[, end]]])](https://nodejs.org/api/buffer.html#buffer_buf_tostring_encoding_start_end) does.

If we want to add all of the different encodings from that csv it should be pretty straightforward once both implementations support the `data-encoding` argument.

Reply:
Thanks @achingbrain!

This PR reminded me of [yEnc](https://en.wikipedia.org/wiki/YEnc) from Usenet days, so I was just looking at efficiency column [in encoding standards comparison](https://en.wikipedia.org/wiki/Binary-to-text_encoding#Encoding_standards) and things like [Ascii85](https://en.wikipedia.org/wiki/Ascii85) (used in PDFs and git patch) looked interesting (due to moderate adoption and smaller overhead, eg.  25% instead of 33%).

..but then i remembered that HTTP API is often gzipped which means any gains introduced by more exotic encoding than Base64 go away.

For example: [base-64 seems to compress better than base-122](http://blog.kevinalbs.com/base122#experimental_results) :upside_down_face: